File: problem/Longest_Subarray/longestSubarray.c
int longestSubarray(int *nums, int numsSize)
{
    if (numsSize == 0)
        return 0;

    int maxVal = nums[0];
    for (int i = 1; i < numsSize; i++)
    {
        if (nums[i] > maxVal)
        {
            maxVal = nums[i];
        }
    }

    int maxLen = 0, currentLen = 0;
    for (int i = 0; i < numsSize; i++)
    {
        if (nums[i] == maxVal)
        {
            currentLen++;
        }
        else
        {
            if (currentLen > maxLen)
            {
                maxLen = currentLen;
            }
            currentLen = 0;
        }
    }

    if (currentLen > maxLen)
    {
        maxLen = currentLen;
    }

    return maxLen;
}

File: problem/Longest_Subarray/longestSubarray.py
class Solution:
    def longestSubarray(self, nums):
        if not nums:
            return 0
            
        max_val = max(nums)
        max_len = 0
        current_len = 0
        
        for num in nums:
            if num == max_val:
                current_len += 1
            else:
                max_len = max(max_len, current_len)
                current_len = 0

        return max(max_len, current_len)
